### DISCUSSION TITLE
How to load custom profile in Chrome using Python Selenium Webdriver?

### DISCUSSION THREAD

**Discussion by 0xlightfury** [community/community#42403] [Programming Help]
### Select Topic Area

Product Feedback

### Body

Here is code
```
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
import time


options = webdriver.ChromeOptions()
options.add_argument("--profile-directory=Profile 6")
options.add_argument("--user-data-dir=C:/Users/SIM/AppData/Local/Google/Chrome/User Data")
driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()), options=options)
time.sleep(3)
```

Here is output
```
D:\me\web\fbot\src>python script_.py
Opening in existing browser session.
[7488:16428:1223/013230.522:ERROR:broker_win.cc(56)] Error reading broker pipe: The pipe has been ended. (0x6D)
Traceback (most recent call last):
  File "D:\me\web\fbot\src\script_.py", line 10, in <module>
    driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()), options=options)
  File "C:\Python310\lib\site-packages\selenium\webdriver\chrome\webdriver.py", line 81, in __init__      
    super().__init__(
  File "C:\Python310\lib\site-packages\selenium\webdriver\chromium\webdriver.py", line 106, in __init__   
    super().__init__(
  File "C:\Python310\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 288, in __init__     
    self.start_session(capabilities, browser_profile)
  File "C:\Python310\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 381, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "C:\Python310\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 444, in execute
    self.error_handler.check_response(response)
  File "C:\Python310\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 249, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: Chrome failed to start: exited normally.
  (unknown error: DevToolsActivePort file doesn't exist)
  (The process started from chrome location C:\Program Files\Google\Chrome\Application\chrome.exe is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
Stacktrace:
Backtrace:
        (No symbol) [0x0103F243]
        (No symbol) [0x00FC7FD1]
        (No symbol) [0x00EBD04D]
        (No symbol) [0x00EDC24E]
        (No symbol) [0x00ED82E9]
        (No symbol) [0x00F0F056]
        (No symbol) [0x00F0EB2A]
        (No symbol) [0x00F08386]
        (No symbol) [0x00EE163C]
        (No symbol) [0x00EE269D]
        GetHandleVerifier [0x012D9A22+2655074]
        GetHandleVerifier [0x012CCA24+2601828]
        GetHandleVerifier [0x010E8C0A+619850]
        GetHandleVerifier [0x010E7830+614768]
        (No symbol) [0x00FD05FC]
        (No symbol) [0x00FD5968]
        (No symbol) [0x00FD5A55]
        (No symbol) [0x00FE051B]
        BaseThreadInitThunk [0x7762FA29+25]
        RtlGetAppContainerNamedObjectPath [0x77CD7A7E+286]
        RtlGetAppContainerNamedObjectPath [0x77CD7A4E+238]
```

Please help me to solve this issue.

**Comment by PeraSite** [comment]
Did you fix it? I have same problem.

**Comment by jfullstackdev** [comment]
not sure of these links 

https://stackoverflow.com/questions/58761312/how-do-i-correctly-use-the-custom-profile-with-selenium-chrome-webdriver-python

https://www.tutorialspoint.com/how-to-use-a-specific-chrome-profile-in-selenium

**Comment by orlandobianco** [comment]
I've the same issue on a fresh installed test setup
I use the same repo (so same settings) in other systems and it works! I'm on windows 10 currently on organization owned device.
all my colleagues use the same "requirements.txt" for the VENV and the same files, but if I try to start a test I get the same error as you.
I don't think I have to change settings on the test setup, cause it won't explain why on other identical machines works well

I've tried a new venv with the newest libraries but same issue, I've tried reinstalling chrome, even the beta and an old release (yeah sure with the right chrome driver for them)
I've the same exact error as you....
are you on windows?

**Reply by 0xlightfury** [reply]
Yes, I am using windows 10

**Reply by orlandobianco** [reply]
have you done some updates recently?
I'm trying to figure out the root of the issue

**Reply by 0xlightfury** [reply]
not yet.

**Reply by orlandobianco** [reply]
are you sure? check in windows updates if there are recent updates
just to be sure
Windows Update --> View Update History: check under all categories like "quality updates" "feature updates" etc..

**Reply by 0xlightfury** [reply]
update is done on 04/19/2022

**Reply by orlandobianco** [reply]
are you using a VENV?

**Reply by 0xlightfury** [reply]
no, I am using VScode

**Reply by orlandobianco** [reply]
ok but are you on a user/global installation of python or in a VENV (virtual env)?
i will open a ticket in the official chrome driver issue tracker

**Reply by orlandobianco** [reply]
I've started this issue 
https://bugs.chromium.org/p/chromedriver/issues/detail?id=4310

**Comment by DeLaphante** [comment]
Struggling to design or build an advanced, stable and well structured test automation framework?

Check out & play around with my powerful and robust Selenium Web UI & API test automation framework on Github -> https://github.com/DeLaphante/CynkyAutomation

Save yourself some time and use this ready made framework for any project and get the technical support you need to succeed!

**Comment by mfrasoulnejad** [comment]
This is related to Chrome browser itself. on MacOs I had to quit the chrome browser first and run my script. that would solve the issue. on my windows PC I had to close all open chrome windows and then run the script. its a little bit strange but that's a workaround!

**Comment by akcilahmet** [comment]
Hello, I am sharing the error code below. Can you help me where am I doing wrong?
![image](https://github.com/community/community/assets/83040124/214cd016-fb13-40e5-9dec-5ce519df1f62)
![image](https://github.com/community/community/assets/83040124/e7f0fc91-b295-4eb3-a789-a9127692e39f)
![image](https://github.com/community/community/assets/83040124/a10ce0ae-1978-45cf-9e53-2104c29393ce)

**Comment by minooza** [comment]
Bro did you found the solution? I got the same problem.

**Reply by akcilahmet** [reply]
Bro, I provided a solution like this, I recommend you try it. I hope it works for you
![image](https://github.com/community/community/assets/83040124/6c464ef2-e8d7-4b9c-ac78-03a4e4fca3c1)

**Reply by MarouaneBenbetka** [reply]
hello try using this package it works fine for me :
```
import undetected_chromedriver as uc
PATH_CHROME_PROFILE = r"__PRIVATE_INFO_HHH_"
options = uc.ChromeOptions()
options.add_argument(
    r"--user-data-dir="+PATH_CHROME_PROFILE)

```

**Comment by MarouaneBenbetka** [comment]
hello try using this package it works fine for me :
```
import undetected_chromedriver as uc
PATH_CHROME_PROFILE = r"__PRIVATE_INFO_HHH_"
options = uc.ChromeOptions()
options.add_argument(
    r"--user-data-dir="+PATH_CHROME_PROFILE)

```

**Comment by Keoush** [comment]
Close all instances of chrome then run selenium script.

**Reply by aliwariss** [reply]
I want to run selenium script while opening all my other chrome tabs because I need those. Is there any fix for this issue. Need help.😥

**Comment by Dev-LK7** [comment]
Olá!
Parece que o erro principal está relacionado ao Chrome não conseguindo iniciar corretamente. Tenho algumas possíveis soluções para esse problema:

**Verifique o caminho do perfil**

Certifique-se de que --user-data-dir=C:/Users/SIM/AppData/Local/Google/Chrome/User Data está correto. Se o perfil não existir ou estiver corrompido, isso pode causar falhas no Chrome.

**Adicione a opção** --remote-debugging-port

O erro menciona que DevToolsActivePort file doesn't exist, então tente adicionar esta linha às opções:

No python coloque essa solução:
options.add_argument("--remote-debugging-port=9222")

**Execute o Chrome manualmente**

Antes de rodar o script, abra manualmente o Chrome com o perfil desejado e veja se ele inicia corretamente.

**Tente sem perfil de usuário**

Remova estas linhas temporariamente e teste:

Python
options.add_argument("--profile-directory=Profile 6")
options.add_argument("--user-data-dir=C:/Users/SIM/AppData/Local/Google/Chrome/User Data")

**Atualize WebDriver e Chrome**

Certifique-se de que seu ChromeDriver está atualizado com pip install --upgrade webdriver-manager.

Certifique-se de que sua versão do Google Chrome corresponde à versão do ChromeDriver instalada.

**Reinicie a máquina**

Algumas vezes, reiniciar o sistema pode ajudar a resolver problemas de inicialização do ChromeDriver.

Testa essas soluções!
Espero ter ajudado!
